### PR TITLE
Fix for #818

### PR DESCRIPTION
**Pull request by badlogic**
The way we implemented `ObjectOwnershipHelper` was incorrect. now search for the first native super class of a custom objc class. In the retain/release callbacks we add the object to our map, then call into retain/release of the native super class.

### CODE REVIEW

File: objc/src/main/java/org/robovm/objc/ObjCObject.java [hunk 1]
```diff
@@ -356,18 +358,39 @@ private static void registerCallbackMethod(long cls, long selector, long newSele
...
-                    CUSTOM_OBJECTS.put(self.getHandle(), self);
+                    ObjCClass cls = ObjCClass.toObjCClass(ObjCRuntime.object_getClass(self));
+                    ObjCObject obj = ObjCObject.toObjCObject(cls.getType(), self, 0);
+                    CUSTOM_OBJECTS.put(self, obj);
                 }
-            }
-            return ObjCRuntime.ptr_objc_msgSendSuper(new Super(self.getHandle(), NS_OBJECT_CLASS).getHandle(), sel);
+            }     
+            long cls = ObjCRuntime.object_getClass(self);
+            long nativeSuper = customClassToNativeSuper.get(cls);
```
Comment (ntherning) [inline objc/src/main/java/org/robovm/objc/ObjCObject.java]: Should the `customClassToNativeSuper.get()` call be in a synchronized block?

File: objc/src/main/java/org/robovm/objc/ObjCObject.java [hunk 2]
```diff
@@ -376,8 +399,11 @@ private static void release(@Pointer long self, @Pointer long sel) {
                 synchronized (CUSTOM_OBJECTS) {
                     CUSTOM_OBJECTS.remove(self);
                 }
-            }
-            ObjCRuntime.void_objc_msgSendSuper(new Super(self, NS_OBJECT_CLASS).getHandle(), sel);
+            }            
+            long cls = ObjCRuntime.object_getClass(self);
+            long nativeSuper = customClassToNativeSuper.get(cls);
```
Comment (ntherning) [inline objc/src/main/java/org/robovm/objc/ObjCObject.java]: Should the `customClassToNativeSuper.get()` call be in a synchronized block?

File: objc/src/main/java/org/robovm/objc/ObjCObject.java [hunk 3]
```diff
@@ -322,7 +322,7 @@ public ObjCObjectRef(ObjCObject referent) {
         }
     }
 
-    static class ObjectOwnershipHelper {
+    public static class ObjectOwnershipHelper {
```
Comment (ntherning) [inline objc/src/main/java/org/robovm/objc/ObjCObject.java]: Why `public`?

**Comment by badlogic** [comment]
Fix things up, ready for merge (also, you are still DND :)).

**Pull request merged**